drivers/storage/portworx/portworx.go

```diff
@@ -736,6 +742,23 @@ func setNodeSpecDefaults(toUpdate *corev1.StorageCluster) {
 	toUpdate.Spec.Nodes = updatedNodeSpecs
 }
 
+func setCSISpecDefaults(toUpdate *corev1.StorageCluster) {
+	if toUpdate.Spec.CSI == nil {
+		toUpdate.Spec.CSI = &corev1.CSISpec{
+			// Enabled by feature gate, but not spec
+			Enabled:                   pxutil.FeatureCSI.IsEnabled(toUpdate.Spec.FeatureGates),
+			InstallSnapshotController: boolPtr(false),
+			InstallSnapshotCRDs:       boolPtr(false),
+		}
+	}
+
+	if toUpdate.Spec.CSI.Enabled && toUpdate.Spec.CSI.InstallSnapshotController == nil {
+		// Do not install snapshot controller by default
+		toUpdate.Spec.CSI.InstallSnapshotController = boolPtr(false)
```

Comment:
I think we can leave it to empty. Anyway we consider empty as false by default. Wdyt?

Reply:
sure, will leave empty